gf-webapp/src/logic/rooms/roomPage/RoomPage.tsx

```diff
@@ -30,19 +30,22 @@ export const RoomPage = () => {
     setRoom(currentRoom);
   };
 
+  const [isRoomInit, setIsRoomInit] = useState(false);
+
   useEffect(() => {
     if (id) {
       loadRoom(id);
+      setIsRoomInit(true);
```

Comment:
If loadRoom is async function, setIsRoomInit(true) will be executed immediately (before loadRoom).

Reply:
Isn't that what's needed? If setIsRoomInit(true) is executed after loadRoom, the Loader will be displayed after the rooms are loaded.

If I write it like this, it will be the same
![Снимок экрана 2023-04-13 114436](https://user-images.githubusercontent.com/103531486/231678764-0d6e217f-9de6-4718-a269-c3c6a7a20ee5.png)

If I call these functions without useEffect then I get errors

If I write it like this then the Loader will be loaded after the loading of the rooms
![Снимок экрана 2023-04-13 115847](https://user-images.githubusercontent.com/103531486/231679175-0b4c8feb-8e66-4695-a380-30169ddb4c79.png)

Reply:
Explain loader visibility logic